build.rs

```diff
@@ -1,3 +1,74 @@
 #![allow(clippy::multiple_crate_versions)]
+use std::env;
+use std::fs::read_dir;
+use std::io::{Error, ErrorKind, Result};
+use std::path::{Path, PathBuf};
 
-fn main() {}
+fn generate_ts_bindings(ts_include_dir: String) -> Result<()> {
+    let header = ts_include_dir.clone() + "/service/locator/service_locator.h";
+
+    println!("cargo:rerun-if-changed={}", header);
+
+    let bindings = bindgen::Builder::default()
+        .clang_arg(format!("-I{}", ts_include_dir))
+        .rustfmt_bindings(true)
+        .header(header)
+        .generate_comments(false)
+        .size_t_is_usize(true)
+        .generate()
+        .or_else(|_| {
+            Err(Error::new(
+                ErrorKind::Other,
+                "Unable to generate bindings to trusted services locator",
+            ))
+        })?;
+    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
+    bindings.write_to_file(out_path.join("ts_bindings.rs"))?;
+    // TODO: Remove once we can use the full TS stack and this isn't needed
+    println!("cargo:rustc-link-search=native=/usr/lib");
+    println!("cargo:rustc-link-lib=dylib=c++");
+
+    println!("cargo:rustc-link-search=native=/usr/local/lib");
+    println!("cargo:rustc-link-lib=static=ts-lib");
+    // TODO: Remove once we can use the full TS stack and this isn't needed
+    println!("cargo:rustc-link-lib=static=mbedcrypto");
+    println!("cargo:rustc-link-lib=static=protobuf-nanopb");
```

Comment:
I'll need to figure out how to make these more sensible

Reply:
Actually, I think I'll leave them as they are - we'll need to remove them before we properly release this provider anyway so they'll only be used for testing